src/lib/uTop.ml

```diff
@@ -33,7 +35,10 @@ let version = "%%VERSION%%"
    +-----------------------------------------------------------------+ *)
 
 let history = LTerm_history.create []
-let history_file_name = ref (Some (Filename.concat LTerm_resources.home ".utop-history"))
+let unopt = function
+        | None -> "None"
+        | Some(v) -> v
+let history_file_name = ref(Some(Filename.concat (unopt(M.cache_dir)) ".utop_history"))
```

Comment:
Using `"None"` as a directory name is probably not what should be done. You should either fail if `M.cache_dire` is `None` (this happens only if there's no way to guess the user's home and directories try quite hard to guess it, so that almost never happen in practice, I've never seen it actually) or just use the current directory and print a warning or whatever.